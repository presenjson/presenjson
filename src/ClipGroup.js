import React, { Component } from 'react';
import Clip from './Clip';
import Scene from './Scene';

export const Group = ({ map, ...props }) => {
    return map.map((x) => <Clip {...props} {...x} />);
};

Group.defaultProps = {
    map: []
};

class ClipGroup extends Component {
    render = () => false
}

ClipGroup.defaultProps = {
    length: 0,
    delay: 0
};

export default ClipGroup;
