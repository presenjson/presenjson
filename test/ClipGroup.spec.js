import React from 'react';
import { shallow } from 'enzyme';
import ClipGroup, { Group } from '../src/ClipGroup';

describe('ClipGroup', () => {
    it('renders nothing', () => {
        expect(
            shallow(
                <ClipGroup>
                    <div className="test" />
                </ClipGroup>
            )
        ).toMatchSnapshot();
    });
    it('sets default props', () => {
        expect(ClipGroup.defaultProps).toEqual({ length: 0, delay: 0 });
    });
    describe('Group', () => {
        it('renders Clip array', () => {
            const Component = () => false;
            const array = [
                {
                    day: 1,
                    description: 'Foo'
                },
                {
                    day: 2,
                    description: 'bar'
                }
            ];
            const group = shallow(
                <Group
                    map={array}
                    some="prop"
                    component={Component}
                    length={1000}
                />
            );
            expect(group).toMatchSnapshot();
        });
    });
});
