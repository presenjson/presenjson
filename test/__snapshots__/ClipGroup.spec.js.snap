// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClipGroup Group renders Clip array 1`] = `
Array [
  <Clip
    component={[Function]}
    delay={0}
    groupData={
      Object {
        "day": 1,
        "description": "Foo",
      }
    }
    key="0"
    length={1000}
    some="prop"
  />,
  <Clip
    component={[Function]}
    delay={0}
    groupData={
      Object {
        "day": 2,
        "description": "bar",
      }
    }
    key="1"
    length={1000}
    some="prop"
  />,
]
`;

exports[`ClipGroup renders nothing 1`] = `""`;
